storagenode/storagenodedb/orders.go

```diff
@@ -113,3 +115,148 @@ func (db *ordersdb) ListUnsent(ctx context.Context, limit int) (_ []*orders.Info
 
 	return infos, ErrInfo.Wrap(rows.Err())
 }
+
+// ListUnsentBySatellite returns orders that haven't been sent yet grouped by satellite.
+// Does not return uplink identity.
+func (db *ordersdb) ListUnsentBySatellite(ctx context.Context) (map[storj.NodeID][]*orders.Info, error) {
+	defer db.locked()()
+	// TODO: add some limiting
+
+	rows, err := db.db.Query(`
+		SELECT order_limit_serialized, order_serialized
+		FROM unsent_order
+	`)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+		return nil, ErrInfo.Wrap(err)
+	}
+	defer func() { err = errs.Combine(err, rows.Close()) }()
+
+	infos := map[storj.NodeID][]*orders.Info{}
+	for rows.Next() {
+		var limitSerialized []byte
+		var orderSerialized []byte
+
+		err := rows.Scan(&limitSerialized, &orderSerialized)
+		if err != nil {
+			return nil, ErrInfo.Wrap(err)
+		}
+
+		var info orders.Info
+		info.Limit = &pb.OrderLimit2{}
+		info.Order = &pb.Order2{}
+
+		err = proto.Unmarshal(limitSerialized, info.Limit)
+		if err != nil {
+			return nil, ErrInfo.Wrap(err)
+		}
+
+		err = proto.Unmarshal(orderSerialized, info.Order)
+		if err != nil {
+			return nil, ErrInfo.Wrap(err)
+		}
+
+		infos[info.Limit.SatelliteId] = append(infos[info.Limit.SatelliteId], &info)
+	}
+
+	return infos, ErrInfo.Wrap(rows.Err())
+}
+
+// Archive marks order as being handled.
+func (db *ordersdb) Archive(ctx context.Context, satellite storj.NodeID, serial storj.SerialNumber, status orders.Status) error {
+	defer db.locked()()
+
+	result, err := db.db.Exec(`
```

Comment:
maybe it's not a very important case here but shouldn't we start using prepared statements for often executed statements?

Reply:
Maybe, let's make an issue about testing the perf difference and see whether it's worthwhile.

Reply:
created an issue

Reply:
thanks